### DISCUSSION TITLE
Can't extract masks for this pdf

### DISCUSSION THREAD

**Discussion by foranuj** [pymupdf/PyMuPDF#2085] [Looking for help] [answered]
I have a PDF that contains four images and I can use the apis to retrieve all the necessary image information as well as manage to extract the full images. 

However, I am not able to retrieve the mask information for the page.

doc.get_page_images(0)
[(6, 0, 4772, 3183, 8, 'ICCBased', '', 'Im1', 'DCTDecode'), (7, 0, 4032, 2688, 8, 'ICCBased', '', 'Im2', 'DCTDecode'), (8, 0, 4032, 2688, 8, 'ICCBased', '', 'Im3', 'DCTDecode'), (9, 0, 4032, 2688, 8, 'ICCBased', '', 'Im4', 'DCTDecode')]

Each of these images are suggesting that the smask is 0.

[sample4.pdf](https://github.com/pymupdf/PyMuPDF/files/10097369/sample4.pdf)

Wondering what am I missing.

**Comment by JorjMcKie** [comment, marked as answer]
An smask value of 0 means there exists not SMask.
Which relates to the fact, that xref numbers always are > 0.
